### PR TITLE
Submit Preflight container cert only if all components are GA

### PR DESCRIPTION
**Pull request by tkrishtop**
Depends-On: https://softwarefactory-project.io/r/c/dci-openshift-app-agent/+/30714

### CODE REVIEW

File: roles/preflight/tasks/test_preflight_check_container_one_image.yml [hunk 1]
```diff
@@ -76,7 +76,7 @@
             {{ preflight_image }}
             check container
             {{ current_operator_image }}
-            {% if cert_project_id | default('') | length %}
+            {% if cert_project_id | default('') | length and dci_all_components_are_ga | default('') | length %}
```
Comment (fredericlepied) [inline roles/preflight/tasks/test_preflight_check_container_one_image.yml:79]: shouldn't it be:
```Python
and dci_all_components_are_ga | default(True)
```

**Comment by tkrishtop** [comment]
recheck

File: roles/preflight/tasks/test_preflight_check_container_one_image.yml [hunk 2]
```diff
@@ -76,7 +76,7 @@
             {{ preflight_image }}
             check container
             {{ current_operator_image }}
-            {% if cert_project_id | default('') | length %}
+            {% if cert_project_id | default('') | length and dci_all_components_are_ga | default(True) %}
```
Comment (thekad) [inline roles/preflight/tasks/test_preflight_check_container_one_image.yml:79]: can you change this to
`if (cert_project_id | default('') | length) and (dci_all_components_are_ga | default(True))`
I feel is more readable.

**Comment by tkrishtop** [comment]
check workload continuous-cert

File: roles/preflight/tasks/test_preflight_check_container_one_image.yml (hunk 2 above)
Comment (tkrishtop) [reply roles/preflight/tasks/test_preflight_check_container_one_image.yml:79]: Hi @thekad thanks for checking. Tbh, I believe that the current way is more conventional and readable, and I'd prefer to keep it.

**Comment by tkrishtop** [comment]
The change is tested:

- Without dallas-pipeline in the depends-zone, the preflight command is executed with `--submit`: https://www.distributed-ci.io/jobs/dbbfd27f-ca90-4deb-994e-dd729dc1ae4d/jobStates?sort=date&task=2934800c-79ea-4f8c-acb6-974c15b4d856
- Tests on GA-OCP and non-GA others (app-agent, dallas-pipelines, dci-pipeline): there is no `--submit`: https://www.distributed-ci.io/jobs/e89068ce-e2f5-46fc-bd62-75ed157542e6/jobStates?sort=date&task=d12eb141-7f89-4ac9-8f8f-06d498c2103d 

These tests show that can never submit because of non-GA app-agent dependency. I'll put PR back into draft mode to think a bit and run more tests.

File: roles/preflight/tasks/test_preflight_check_container_one_image.yml (hunk 1 above)
Comment (tkrishtop) [reply roles/preflight/tasks/test_preflight_check_container_one_image.yml]: fixed, thanks Fred and Pierre!

**Comment by tkrishtop** [comment]
check workload continuous-cert

**Comment by tkrishtop** [comment]
check workload continuous-cert

**Comment by tkrishtop** [comment]
check workload continuous-cert

File: roles/preflight/tasks/test_preflight_check_container_one_image.yml (hunk 2 above)
Comment (tonyskapunk) [inline roles/preflight/tasks/test_preflight_check_container_one_image.yml:79]: Please consider renaming (with `preflight_` prefix) this var and documenting it

Comment (tkrishtop) [reply roles/preflight/tasks/test_preflight_check_container_one_image.yml:79]: Hi @tonyskapunk thanks for checking, I'm far from merging here. 
Somehow the OCP component on cluster10 where we have the latest GA [is not detected as a GA component](https://www.distributed-ci.io/jobs/a10685c6-442e-4f5f-83d0-9239390a05fb/jobStates?sort=date&task=bc33dc68-b1f0-477e-85d0-3aa0630b12b0):
```
task path: /home/dciteam/github/ansible-collection-redhatci-ocp-pr139-53ab370dfde4580c39d8e08339915c5f/dci-openshift-app-agent/plays/pre-run.yml:40
ok: [jumphost] => {
    "msg": [
        "dci_ga_components_for_certification: ['ocp']",
        "ga_components: []",
        "dci_all_components_are_ga: False"
    ]
}
```
Have to understand and fix that first.

Comment (tkrishtop) [reply roles/preflight/tasks/test_preflight_check_container_one_image.yml:79]: This component has `build:ga` tag but the query somehow does not pick it up
```
ga_components: "{{ job_info['job']['components'] | selectattr('tags', 'issuperset', ['build:ga']) | map(attribute='type') | list }}"
```
I will debug it tomorrow, I don't see the issue with the query right now :(

**Comment by tkrishtop** [comment]
check workload continuous-cert

File: roles/preflight/tasks/test_preflight_check_container_one_image.yml (hunk 2 above)
Comment (fredericlepied) [reply roles/preflight/tasks/test_preflight_check_container_one_image.yml:79]: can you display the components from the job?

**Comment by fredericlepied** [comment]
check workload continuous-cert

**Comment by fredericlepied** [comment]
check workload continuous-cert

**Comment by fredericlepied** [comment]
check workload continuous-cert

**Comment by fredericlepied** [comment]
check workload continuous-cert

**Comment by fredericlepied** [comment]
check workload continuous-cert

**Comment by fredericlepied** [comment]
check workload continuous-cert

**Comment by fredericlepied** [comment]
check workload continuous-cert

**Comment by tkrishtop** [comment]
check workload continuous-cert

**Comment by tkrishtop** [comment]
check workload continuous-cert

**Comment by tkrishtop** [comment]
The change is tested fine, the run on cluster10 with GA OCP shows the [--submit to be added](https://www.distributed-ci.io/jobs/4f5484ee-cea4-4341-b493-99781af4ae98/jobStates?sort=date&task=72e9df6f-cda5-45f2-b57e-f7293f6d90c4) when [dci_all_components_are_ga is True](https://www.distributed-ci.io/jobs/4f5484ee-cea4-4341-b493-99781af4ae98/jobStates?sort=date&task=3a6555c6-22c9-483f-af8b-493479ae8f07).

File: roles/preflight/tasks/test_preflight_check_container_one_image.yml (hunk 2 above)
Comment (tkrishtop) [reply roles/preflight/tasks/test_preflight_check_container_one_image.yml]: @tonyskapunk I added the changes you have requested. Could you please have a look one more time?

**Review by ramperher — APPROVED** [review]
LGTM

**Pull request merged**